### PR TITLE
LoginPopup for Follow button

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Partially addresses #1051

#### What's this PR do?
- Adds the `LoginPopup` component to the follow button
- Creates a new class for `FollowButton` instead of creating it via a function within ExpandedPostDisplay.
- Adds `className` to the LoginPopup so its styling can be customized (necessary for the Follow button, to avoid it overflowing past the right edge of the browser window).

#### How should this be manually tested?
- When logged in, the Follow/Unfollow button should work as before on the post detail page.
- When not logged in, the Follow button should still display and clicking it should display the login popup.

### CODE REVIEW

**Comment by aliceriot** [comment]
functionality looks great! reading code now

File: static/js/components/ExpandedPostDisplay_test.js [hunk 1]
```diff
@@ -197,23 +197,6 @@ describe("ExpandedPostDisplay", () => {
...
-        const wrapper = renderPostDisplay()
-        const button = wrapper.find(
-          subscribed ? ".subscribed" : ".unsubscribed"
-        )
-        assert.include(button.text(), buttonText)
-        button.simulate("click")
-        assert.ok(toggleFollowPostStub.called)
-      })
-    }
-  )
```
Comment (aliceriot) [inline static/js/components/ExpandedPostDisplay_test.js]: might be good to add a test here just confirming that the right props are passed down to the `FollowButton` component

File: static/js/components/FollowButton_test.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import { shouldIf } from "../lib/test_utils"
+import * as utilFuncs from "../lib/util"
+import Router from "../Router"
+import IntegrationTestHelper from "../util/integration_test_helper"
+
+describe("FollowButton", () => {
+  let helper, toggleFollowPostStub, post
+
+  beforeEach(() => {
+    //sandbox = sinon.createSandbox()
```
Comment (aliceriot) [inline static/js/components/FollowButton_test.js]: stray comment?

File: static/js/components/LoginPopup.js [hunk 3]
```diff
@@ -1,14 +1,15 @@
 // @flow
-import { Link } from "react-router-dom"
 import React from "react"
 import onClickOutside from "react-onclickoutside"
 
 import Button from "./Button"
+import { Link } from "react-router-dom"
```
Comment (aliceriot) [inline static/js/components/LoginPopup.js]: small thing, but we usually keep outside dependencies together at the top

File: static/scss/share-popup.scss [hunk 4]
```diff
@@ -46,6 +46,14 @@
   }
 }
```
Comment (aliceriot) [inline static/scss/share-popup.scss]: this file is styling both the share and login popups I think, what do you think about renaming it to just `popup.scss` while we're in here? I figure that's a little more clear

Comment (mbertrand) [reply static/scss/share-popup.scss]: 👍

File: static/js/components/ExpandedPostDisplay_test.js (hunk 1 above)
Comment (mbertrand) [reply static/js/components/ExpandedPostDisplay_test.js]: ✅

**Pull request merged**